aidies-project/client/src/App.js

```diff
@@ -1,35 +1,49 @@
-import React from 'react';
-import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
-import {  useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { BrowserRouter, Switch, Route, Redirect, useLocation } from 'react-router-dom';
+import {  useSelector, useDispatch } from 'react-redux';
+
+import { setLocation} from './store/campaign';
 
 import Navigation from './components/Navigation';
 import LoginPanel from './components/LoginPanel';
 import SignupPanel from './components/SignupPanel';
 import CampaignBrowser from './components/CampaignBrowser';
 import CampaignDetail from './components/CampaignDetail';
 import CampaignCreate from './components/CampaignCreate';
+import CampaignEdit from './components/CampaignEdit';
+import Dashboard from './components/Dashboard';
+import Footer from './components/Footer';
+
 
+const PrivateRoute = ({component: Component, ...rest}) => {
+  const dispatch = useDispatch();
+  const location = useLocation();
+  useEffect(()=>{
+    dispatch(setLocation(location.pathname));
+  }, [location.pathname])
```

Comment:
I think this might be tied with your infinite loop problem! I feel like the `setLocation` action should only be dispatched when a component initially mounts, rather than when the `location.pathname` updates.

```suggestion
  }, [])
```